src/msg/async/rdma/RDMAStack.cc

```diff
@@ -244,15 +244,18 @@ void RDMADispatcher::polling()
       perf_logger->set(l_msgr_rdma_inflight_tx_chunks, inflight);
       if (num_dead_queue_pair) {
         Mutex::Locker l(lock); // FIXME reuse dead qp because creating one qp costs 1 ms
-        while (!dead_queue_pairs.empty()) {
-          ldout(cct, 10) << __func__ << " finally delete qp=" << dead_queue_pairs.back() << dendl;
-          delete dead_queue_pairs.back();
+        for (auto idx = 0; idx < dead_queue_pairs.size(); idx++) {
```

Comment:
use range-based loop please to avoid repeating `dead_queue_pairs.at(idx)`.

Reply:
Do you want me to use the following style? Actually I am new with C++ 11, pls correct me if anything is incorrect; thanks!        

        int idx = 0;
        for (auto &i : dead_queue_pairs) {
                idx++;
                if (i->get_tx_wc() != i->get_tx_wr())
                        continue;
                ldout(cct, 10) << __func__ << "finally delete qp=" << i << dendl;
                delete i;
                dead_queue_pairs.erase(dead_queue_pairs.begin() + idx - 1);
                perf_logger->dec(l_msgr_rdma_active_queue_pair);
                --num_dead_queue_pair;
        }